diesel/src/sql_types/mod.rs

```diff
@@ -349,6 +349,41 @@ pub struct Time;
 #[diesel(mysql_type(name = "Timestamp"))]
 pub struct Timestamp;
 
+/// The timestamp with timezone SQL type.
+///
+/// ### [`ToSql`](crate::serialize::ToSql) impls
+///
+/// - [`std::time::SystemTime`][SystemTime] (PG only)
+/// - [`chrono::NaiveDateTime`][NaiveDateTime] with `feature = "chrono"`
+/// - [`time::Timespec`][Timespec] with `feature = "deprecated-time"` (PG only)
+///
+/// ### [`FromSql`](crate::deserialize::FromSql) impls
+///
+/// - [`std::time::SystemTime`][SystemTime] (PG only)
+/// - [`chrono::NaiveDateTime`][NaiveDateTime] with `feature = "chrono"`
+/// - [`time::Timespec`][Timespec] with `feature = "deprecated-time"` (PG only)
+///
+/// [SystemTime]: std::time::SystemTime
+#[cfg_attr(
+    feature = "chrono",
+    doc = " [NaiveDateTime]: chrono::naive::NaiveDateTime"
+)]
+#[cfg_attr(
+    not(feature = "chrono"),
+    doc = " [NaiveDateTime]: https://docs.rs/chrono/*/chrono/naive/struct.NaiveDateTime.html"
+)]
+#[cfg_attr(feature = "chrono", doc = " [`chrono::DateTime`]: chrono::DateTime")]
+#[cfg_attr(
+    not(feature = "chrono"),
+    doc = " [`chrono::DateTime`]: https://docs.rs/chrono/0.4.19/chrono/struct.DateTime.html"
+)]
+/// [Timespec]: /time/struct.Timespec.html
+#[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
+#[diesel(postgres_type(oid = 1184, array_oid = 1185))]
+#[diesel(sqlite_type(name = "Text"))]
+#[diesel(mysql_type(name = "Timestamp"))]
+pub struct TimestampWithTz;
```

Comment:
I know that there exists already `Timestamptz` for the postgres backend and I'm guessing that, at least in theory, that should be something of a subclass of this one? But I wasn't sure if that was in scope for this issue or not

Reply:
So I think I understand what you're saying in theory but the Rust module system is still new to me so... I did what I could (as far as I can tell, the MySQL and Postgres backends expose their specific types a little differently) so I copied a wee bit what MySQL did -- but that meant that I didn't actually create any new `sqlite/types.rs` module 😅  

Anyway I'm pushing up a new commit now (tests still aren't passing but at least everything compiles!) so will see what you think from there

Reply:
Pushing up a commit that I think finally makes the whole thing ready for review; resolving and can start a fresh round of feedback